### DISCUSSION TITLE
How do install via docker on a server that has an existing Postgres instance?

### DISCUSSION THREAD

**Discussion by twointum** [apache/superset#12876] [Q&A / Help]
Hello all, I am getting these errors after using the docker-compose up command:

```
Creating superset_db ...
Creating superset_db ... error

ERROR: for superset_db  Cannot start service db: driver failed programming external connectivity on endpoint superset_db (2080276d92973ae2c6744                     7c837241ace508f7a67a4125c6ca8120b183b8495e1): Error starting userland proxy: listen tcp 127.0.0.1:5432: bind: address already in use

ERROR: for db  Cannot start service db: driver failed programming external connectivity on endpoint superset_db (2080276d92973ae2c67447c837241a                     ce508f7a67a4125c6ca8120b183b8495e1): Error starting userland proxy: listen tcp 127.0.0.1:5432: bind: address already in use
ERROR: Encountered errors while bringing up the project.
```

I do have an existing PG instance on this server so that's why 5432 is in use. How do I use a different port for SuperSet's PG instance? Thanks!

**Comment by nytai** [comment]
You'll need to change a few values. 
Change this line to https://github.com/apache/superset/blob/86807e40b778495d3f5155e043fd6f61d787f17e/docker-compose.yml#L46
to `- "127.0.0.1:<your desired port>:5432"`

and this line
https://github.com/apache/superset/blob/86807e40b778495d3f5155e043fd6f61d787f17e/docker/.env#L27
to `DATABASE_PORT=<your desired port>`

**Reply by twointum** [reply]
Getting closer, I don't know what this 172.18.0.3 IP is, I'm running on localhost.

```ERROR:flask_appbuilder.security.sqla.manager:DB Creation and initialization failed: (psycopg2.OperationalError) could not connect to server: Connection refused
 Is the server running on host "db" (172.18.0.3) and accepting
TCP/IP connections on port 5430?

(Background on this error at: http://sqlalche.me/e/13/e3q8)
Loaded your LOCAL configuration at [/app/docker/pythonpath_dev/superset_config.py]
```

Also, do you know how to make it use my pyenv python environment (if necessary)? Thank you so much!

**Reply by nytai** [reply]
Actually, the only change that's necessary is  `"127.0.0.1:<your desired port>:5432"`. Superset is connecting to the db via an internal docker address in most instances.

**Reply by nytai** [reply]
If you don't need to connect to the db directly from outside of docker, you could just comment out that entire line. `superset_tests_worker` will fail to boot because it tries to connect over localhost, so you'd probably also want to comment out that entire section of the docker-compose file.

**Reply by twointum** [reply]
This turned out to be hardware. I had a Core 2 Quad that was apparently too crappy to run this. Thanks for the help!
